cli/cmd/alert_list.go

```diff
@@ -0,0 +1,262 @@
+    B. RFC3339 date and time
+    C. Epoch time in milliseconds
+
+To list open alerts of type "NewViolations" with high or critical severity.
+
+    lacework alert ls --status Open --severity high --type NewViolations
+`,
+		Args: cobra.NoArgs,
+		PreRunE: func(_ *cobra.Command, args []string) error {
+			// validate severity
+			if alertCmdState.Severity != "" && !array.ContainsStr(
+				api.ValidAlertSeverities, alertCmdState.Severity) {
+				return errors.Wrap(
+					errors.New(fmt.Sprintf("the severity (%s) is not valid, use one of (%s)",
+						alertCmdState.Severity, strings.Join(api.ValidAlertSeverities, ", "))),
+					"unable to list alerts",
+				)
+			}
+			// validate status
+			if alertCmdState.Status != "" && !array.ContainsStr(
+				api.ValidAlertStatuses, alertCmdState.Status) {
+				return errors.Wrap(
+					errors.New(fmt.Sprintf("the status (%s) is not valid, use one of (%s)",
+						alertCmdState.Status, strings.Join(api.ValidAlertStatuses, ", "))),
+					"unable to list alerts",
+				)
+			}
+			return nil
+		},
+		RunE: listAlert,
+	}
+)
+
+func init() {
+	alertCmd.AddCommand(alertListCmd)
+
+	// range time flag
+	alertListCmd.Flags().StringVar(
+		&alertCmdState.Range,
+		"range", "",
+		"natural time range for alerts",
+	)
+
+	// start time flag
+	alertListCmd.Flags().StringVar(
+		&alertCmdState.Start,
+		"start", "-24h",
+		"start time for alerts",
+	)
+	// end time flag
+	alertListCmd.Flags().StringVar(
+		&alertCmdState.End,
+		"end", "now",
+		"end time for alerts",
+	)
+
+	// severity flag
+	alertListCmd.Flags().StringVar(
+		&alertCmdState.Severity,
+		"severity", "",
+		fmt.Sprintf(
+			"filter alerts by severity threshold (%s)",
+			strings.Join(api.ValidAlertSeverities, ", "),
+		),
+	)
+
+	// status flag
+	alertListCmd.Flags().StringVar(
+		&alertCmdState.Status,
+		"status", "",
+		fmt.Sprintf(
+			"filter alerts by status (%s)",
+			strings.Join(api.ValidAlertStatuses, ", "),
+		),
+	)
+
+	// type flag
+	alertListCmd.Flags().StringVar(
+		&alertCmdState.Type,
+		"type", "",
+		"filter alerts by type",
+	)
+}
+
+func alertListTable(alerts api.Alerts) (out [][]string) {
+	sevThreshold, _ := severityToProperTypes(alertCmdState.Severity)
+
+	for _, alert := range alerts.SortDescending() {
+		// filter severity if desired
+		if sevThreshold > 0 {
+			alertSeverity, _ := severityToProperTypes(
+				strings.ToLower(alert.Severity),
+			)
+
+			if alertSeverity > sevThreshold {
+				continue
+			}
+		}
+
+		out = append(out, []string{
+			strconv.Itoa(alert.ID),
+			alert.Type,
+			alert.Name,
+			alert.Severity,
+			alert.StartTime,
+			alert.EndTime,
+			alert.Status,
+		})
+	}
+
+	return
+}
+
+func renderAlertListTable(alerts api.Alerts) {
+	cli.OutputHuman(
+		renderCustomTable(
+			[]string{"Alert ID", "Type", "Name", "Severity", "Start Time", "End Time", "Status"},
+			alertListTable(alerts),
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetAutoWrapText(false)
+				t.SetBorder(false)
+			}),
+		),
+	)
```

Comment:
Can we copy the table options from "events2?
```go
// easily add or remove borders to all event details tables
func eventsTableOptions() tableOption {
	return tableFunc(func(t *tablewriter.Table) {
		t.SetBorders(tablewriter.Border{
			Left:   false,
			Right:  false,
			Top:    true,
			Bottom: false,
		})
		t.SetColumnSeparator(" ")
	})
}
```

What do you think?

Reply:
Do you want all the alert tables to use these options?

Reply:
It really depends, I wish we could have someone from UX to look at all these tables. When I did
`lacework event` I tried to use the terminal space as much as possible. I will leave it to your criteria
but use as much space as possible when it makes sense and when it doesn't make the output look
confusing, that is, when there are big long text.

Reply:
I'm going to create something for follow up on this.